encoding/thrift/inbound.go

```diff
@@ -71,7 +72,7 @@ func (t thriftUnaryHandler) Handle(ctx context.Context, treq *transport.Request,
 		}
 	}
 
-	envelope, err := proto.DecodeEnveloped(bytes.NewReader(body))
+	envelope, err := proto.DecodeEnveloped(bytes.NewReader(buf.Bytes()))
```

Comment:
I think you can pass `buf` in as-is instead of building a new reader. bytes.Buffer tracks the read and write positions separately. 

https://play.golang.org/p/vRbajakdAh